From b5bb1debe4f1686aa018cc941389ced8c4b97cbb Mon Sep 17 00:00:00 2001
From: Juan RP <xtraeme@gmail.com>
Date: Mon, 14 Jan 2013 14:01:35 +0100
Subject: [PATCH] Partial revert of 5d59a15faf67d921dfb190195c07bf4a70bca137.

Error out if "pre-remove" script action fails, otherwise package is
removed but its files are still there (ouch).
---
 NEWS                 | 3 ---
 lib/package_remove.c | 5 ++---
 2 files changed, 2 insertions(+), 6 deletions(-)

diff --git a/lib/package_remove.c b/lib/package_remove.c
index a207fc0..4e69e82 100644
--- lib/package_remove.c
+++ lib/package_remove.c
@@ -264,9 +264,8 @@ xbps_remove_pkg(struct xbps_handle *xhp,
 			    errno, pkgname, version,
 			    "%s: [remove] REMOVE script failed to "
 			    "execute pre ACTION: %s",
-			    pkgver, strerror(errno));
-			rv = errno;
-			goto purge;
+			    pkgver, strerror(rv));
+			goto out;
 		}
 	}
 	/*
-- 
1.8.1

